pkg/platform/runtime/artifactcache/artifactcache.go

```diff
@@ -0,0 +1,172 @@
+package artifactcache
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/installation/storage"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+)
+
+type cachedArtifact struct {
+	Id             artifact.ArtifactID `json:"id"`
+	ArchivePath    string              `json:"archivePath"`
+	Size           int64               `json:"size"`
+	LastAccessTime int64               `json:"lastAccessTime"`
+}
+
+// ArtifactCache is a cache of downloaded artifacts from the ActiveState Platform.
+// The State Tool prefers to use this cache instead of redownloading artifacts.
+type ArtifactCache struct {
+	dir         string
+	infoJson    string
+	maxSize     int64 // bytes
+	currentSize int64 // bytes
+	artifacts   map[artifact.ArtifactID]*cachedArtifact
+	mutex       sync.Mutex
+}
+
+const MB int64 = 1024 * 1024
+
+// New returns a new artifact cache in the State Tool's cache directory with the default maximum size of 500MB.
+func New() (*ArtifactCache, error) {
+	var maxSize int64 = 500 * MB
+	if sizeOverride, err := strconv.Atoi(os.Getenv(constants.ArtifactCacheSizeEnvVarName)); err != nil && sizeOverride > 0 {
```

Comment:
We should at least print a warning to the user if they passed an invalid value.

Reply:
Leaving this for implementation with config option as it's rather involved with our event producer -> consumer framework.